sudo/views.py

```diff
@@ -88,10 +89,14 @@ def sudo(request, template_name='sudo/sudo.html', extra_context=None):
     if request.is_sudo():
         return HttpResponseRedirect(redirect_to)
 
+    if request.method == 'GET':
+        request.session['redirect_to'] = redirect_to
+
     form = SudoForm(request.user, request.POST or None)
     if request.method == 'POST':
         if form.is_valid():
             grant_sudo_privileges(request)
+            redirect_to = request.session.pop('redirect_to', redirect_to)
```

Comment:
For paranoia, we should double check the value here and pass it through the `is_safe_url()` again. This way we can assert that the value wasn't changed to something malicious between the GET and POST.